Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -105,6 +105,35 @@ extension Channel {
 
         return self.pipeline.addHandlers(handlers, position: position).map { multiplexer }
     }
+    
+    /// Configures a `ChannelPipeline` to speak HTTP/2.
+    ///
+    /// In general this is not entirely useful by itself, as HTTP/2 is a negotiated protocol. This helper does not handle negotiation.
+    /// Instead, this simply adds the handlers required to speak HTTP/2 after negotiation has completed, or when agreed by prior knowledge.
+    /// Whenever possible use this function to setup a HTTP/2 server pipeline, as it allows that pipeline to evolve without breaking your code.
+    ///
+    /// - parameters:
+    ///     - mode: The mode this pipeline will operate in, server or client.
+    ///     - initialLocalSettings: The settings that will be used when establishing the connection. These will be sent to the peer as part of the
+    ///         handshake.
+    ///     - position: The position in the pipeline into which to insert these handlers.
+    ///     - targetWindowSize: The target size of the inbound flow control window.
+    ///     - inboundStreamStateInitializer: A closure that will be called whenever the remote peer initiates a new stream. This should almost always
+    ///         be provided, especially on servers.
+    /// - returns: An `EventLoopFuture` containing the `HTTP2StreamMultiplexer` inserted into this pipeline, which can be used to initiate new streams.
+    public func configureHTTP2Pipeline(mode: NIOHTTP2Handler.ParserMode,
```

Comment:
Could you have the other implementation call this implementation with the default value of `targetWindowSize`? I'd rather duplicate the default value than the function body.

Reply:
Sure, thought about that, but didn't know who determined the default value -- is the presumption that default value is `SETTINGS_INITIAL_WINDOW_SIZE`'s default of 65535?

I.e., if calling the new from the old with 65535, the default values in `HTTP2StreamMultiplexer` are essentially meaningless since they are never invoked without a `targetWindowSize` arg value.

Reply:
> Sure, thought about that, but didn't know who determined the default value -- is the presumption that default value is `SETTINGS_INITIAL_WINDOW_SIZE`'s default of 65535?

The default should be whatever the multiplexer defaults to.

> I.e., if calling the new from the old with 65535, the default values in `HTTP2StreamMultiplexer` are essentially meaningless since they are never invoked without a `targetWindowSize` arg value.

That's only true if you're using these functions to construct an HTTP2 pipeline: it's still valid to add the HTTP2 handler and multiplexer manually.